### ISSUE TITLE
"reduce" equivalent?

### ISSUE THREAD

**Issue by yozlet** [sanity-io/GROQ#24]
So far, GROQ has `map` (projection) and `filter` covered; how about the missing member of the trinity, `reduce`?

**Comment by judofyr** [comment]
Yes, `reduce` would be very interesting. We've been thinking about aggregate queries in general (`sum`, `avg` and so one) and `reduce` is very related to this. I'm not quite sure how the syntax would look though. Any thoughts?
